### PR TITLE
Planck keymap

### PR DESCRIPTION
**Pull request by lja83**
## Description

Created a new layout for planck
Added a nav layer with mouse controls and arrow keys on hjkl as well as esc/ctrl and return/shift mod taps

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/planck/keymaps/lja83/config.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
+#pragma once
```
Comment (drashna) [inline keyboards/planck/keymaps/lja83/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/planck/keymaps/lja83/config.h [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+/* enable advanced MIDI features:
+   - MIDI notes can be added to the keymap
+   - Octave shift and transpose
+   - Virtual sustain, portamento, and modulation wheel
+   - etc.
+*/
+//#define MIDI_ADVANCED
+
+/* override number of MIDI tone keycodes (each octave adds 12 keycodes and allocates 12 bytes) */
+//#define MIDI_TONE_KEYCODE_OCTAVES 2
```
Comment (fauxpark) [inline keyboards/planck/keymaps/lja83/config.h:48]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by lja83** [comment]
@fauxpark Would you mind reviewing again?

**Review by fauxpark — CHANGES_REQUESTED** [review]
.

**Review by fauxpark — APPROVED** [review]

File: keyboards/planck/keymaps/lja83/keymap.c [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+  return true;
+}
+
+bool muse_mode = false;
+uint8_t last_muse_note = 0;
+uint16_t muse_counter = 0;
+uint8_t muse_offset = 70;
+uint16_t muse_tempo = 50;
+
+void encoder_update(bool clockwise) {
```
Comment (drashna) [inline keyboards/planck/keymaps/lja83/keymap.c:269]:
```suggestion
bool encoder_update_user(uint16_t index, bool clockwise) {
```

File: keyboards/planck/keymaps/lja83/keymap.c [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+      #endif
+    } else {
+      #ifdef MOUSEKEY_ENABLE
+        tap_code(KC_MS_WH_UP);
+      #else
+        tap_code(KC_PGUP);
+      #endif
+    }
+  }
+}
```
Comment (drashna) [inline keyboards/planck/keymaps/lja83/keymap.c:299]:
```suggestion
    return true;
}
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by lja83** [comment]
The changes I'm being asked to make are all things that were in the default files from `qmk new-keymap`.  Why do the default keymap files not meet the code standards?

**Comment by drashna** [comment]
the `encoder_update_user` function has changed, and without the suggested changes, it won't compile, if you update the repo.

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**